e2e/wdio/platform/tests/wizard-generator.e2e-spec.ts

```diff
@@ -0,0 +1,545 @@
+            click(selector + select);
+            click(listItem);
+        }
+        selector === externalNavigationExample || selector === customizableGeneratorExample
+            ? click(selector + nextStepBtn2)
+            : click(selector + nextStepBtn);
+        expect(getElementClass(selector + step, 1)).toContain('current', 'second step is not current');
+        expect(getElementClass(selector + step, 0)).toContain('completed', 'first step is not completed');
+        scrollIntoView(selector + stepContainer);
+        scrollIntoView(selector + step);
+        expect(getElementClass(selector + step, 0)).toContain('current', 'step did not back to first');
+        expect(getElementClass(selector + step, 1)).toContain('upcoming', 'step did not back to first');
+    }
+
+    function checkNavigationByButtonsInDialog(selector: string): void {
+        scrollIntoView(selector);
+        click(selector + button);
+        click(dialog + select);
+        click(listItem);
+        click(dialog + nextStepBtn2);
+        expect(getElementArrayLength(dialogBarButton)).toBe(3, 'back button is appeared');
+        expect(getElementClass(dialog + step, 1)).toContain('current', 'second step is not current');
+        expect(getElementClass(dialog + step, 0)).toContain('completed', 'first step is not completed');
+        click(dialogBarButton);
+        expect(getElementClass(dialog + step, 0)).toContain('current', 'step did not back to first');
+        expect(getElementClass(dialog + step, 1)).toContain('upcoming', 'step did not back to first');
+        expect(getElementArrayLength(dialogBarButton)).toBe(2, 'back button is not hidden');
+    }
+
+    function checkNavigationBySteps(selector: string): void {
+        scrollIntoView(selector);
+        expect(isElementClickable(selector + stepContainer, 1)).toBe(false, 'second step clickable but should not');
+        if (selector === summaryObjectsExample) {
+            click(checkboxLabel);
+        }
+        if (selector === onChangeExample) {
+            setValue(selector + input, name);
+        }
+        if (selector !== summaryObjectsExample && selector !== onChangeExample) {
+            click(selector + select);
+            click(listItem);
+        }
+        selector === externalNavigationExample || selector === customizableGeneratorExample || selector === dialog
+            ? click(selector + nextStepBtn2)
+            : click(selector + nextStepBtn);
+        expect(getElementClass(selector + step, 1)).toContain('current', 'you not moved to second step');
+        expect(getElementClass(selector + step, 0)).toContain('completed', 'first step is not completed');
+        click(selector + stepContainer);
+        expect(getElementClass(selector + step, 0)).toContain(
+            'current',
+            'you did not get back to first step by click on step tab'
+        );
+    }
+
+    function checkValidationRequiredFields(selector: string): void {
```

Comment:
1. better to break this into separate methods 
2. need to add validation message check as well (for example on step 2 if you click on the field after error class applied, popup with text appears - need to assert message and message text
![2021-11-01](https://user-images.githubusercontent.com/47522152/139654494-4cb19287-2499-4417-b9fc-584b9218601a.png)